### PR TITLE
chore(litmus-portal): Added image registry frontend integration

### PR DESCRIPTION
**Pull request by amityt**
Signed-off-by: Amit Kumar Das <amit@chaosnative.com>

## Proposed changes

This PR adds:
- Image registry frontend integration

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

## Special notes for your reviewer:
Screenshot from 2021-05-12 13-47-23

### CODE REVIEW

File: litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+        data.GetImageRegistry.image_registry_info.image_repo_name ===
+        'litmuschaos'
+      ) {
+        setRegistry('disabled');
+        setIsCustomRegistryEnabled(false);
+        setRegistryData({
+          registry_name: 'Docker Hub',
+          repo_name: 'litmuschaos',
+          registry_type: 'Public',
+        });
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx:126]: declare in a constants file (with default values)

File: litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+  /**
+   * UpdateImageRegistry is used to update the Image Registry
+   */
+  const [updateImageRegistry, { loading: updateLoader }] = useMutation(
+    UPDATE_IMAGE_REGISTRY,
+    {
+      onCompleted: (data) => {
+        if (
+          data.updateImageRegistry.image_registry_info.image_repo_name ===
+          'litmuschaos'
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx:91]: constant would be better

File: litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+  const [registryType, setRegistryType] = useState('Public');
+  const [isCustomRegistryEnabled, setIsCustomRegistryEnabled] = useState(false);
+  /**
+   * GetRegistryData to fetch Registry Data by ID
+   */
+  const [getRegistryData, { data, loading }] = useLazyQuery(GET_IMAGE_REGISTRY);
+  const [registryInfo, setRegistryInfo] = useState<RegistryInfo>({
+    registry_name: '',
+    repo_name: '',
+    registry_type: 'Public',
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx:54]: use constant

File: litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+                      disabled={updateLoader}
+                      onClick={() =>
+                        updateImageRegistry({
+                          variables: {
+                            imageRegistryID: registryID,
+                            projectID,
+                            imageRegistryInfo: {
+                              image_registry_name: 'docker.io',
+                              image_repo_name: 'litmuschaos',
+                              image_registry_type: 'Public',
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx:235]: constants needed.

File: litmus-portal/frontend/src/views/Settings/ImageRegistry/styles.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    marginTop: theme.spacing(2.5),
+    display: 'flex',
+  },
+  inputDiv: {
+    marginRight: theme.spacing(5),
+  },
+  formControl: {
+    marginTop: theme.spacing(2.5),
+    width: '12.5rem',
+    marginBottom: theme.spacing(2.5),
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/views/Settings/ImageRegistry/styles.ts:58]: use one margin property

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

**Review by ishangupta-ds — APPROVED** [review]
LGTM

File: litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+        {t('settings.imageRegistry.heading')}
+      </Typography>
+      <Typography className={classes.headerDesc}>
+        {t('settings.imageRegistry.headingDesc')}
+      </Typography>
+
+      {listLoading && loading ? (
+        <>
+          <Loader />
+        </>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx:181]:
```suggestion
      {listLoading && loading ? (
          <Loader />
```

File: litmus-portal/frontend/public/locales/en/translation.yaml [hunk 7]
```diff
@@ -628,6 +628,23 @@ settings:
...
+    repo: 'Repository : '
+    repoType: 'Registry Type : '
+    dockerio: 'docker.io'
+    litmus: 'litmuschaos'
+    public: Public
+    private: Private
+    defaultReg: Use Default Registry
+    customValues: Use custom values
+    registryLabel: Registry Type
+    additionalInfo: Provide addtional details
```
Comment (Jonsy13) [inline litmus-portal/frontend/public/locales/en/translation.yaml:646]: Can we fix additional spelling here

File: litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+                      <Typography className={classes.registryInfoText}>
+                        {t('settings.imageRegistry.repo')}
+                        <strong>{registryData.repo_name}</strong>
+                      </Typography>
+                      <Typography className={classes.registryInfoText}>
+                        {t('settings.imageRegistry.repoType')}
+                        <strong>{registryData.registry_type}</strong>
+                      </Typography>
+                    </div>
+                  </>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/views/Settings/ImageRegistry/index.tsx:291]: Fragments aren't required here, div is already single wrapper

File: litmus-portal/frontend/src/views/Settings/ImageRegistry/styles.ts [hunk 9]
```diff
@@ -0,0 +1,67 @@
+import { makeStyles, Theme } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  container: {
+    marginTop: theme.spacing(3.75),
+  },
+  mainRadioDiv: {
+    marginTop: theme.spacing(2),
+    padding: theme.spacing(3.125),
+    border: `1px solid ${theme.palette.border.main}`,
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/views/Settings/ImageRegistry/styles.ts:10]: Could use rem instead for consistency, not an issue though

**Review by S-ayanide — APPROVED** [review]

**Review by Jonsy13 — APPROVED** [review]

**Review by S-ayanide — APPROVED** [review]

**Pull request merged**